engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -249,23 +236,36 @@ private void preRenderUpdate(DefaultRenderingProcess.StereoRenderState stereoRen
             renderableWorld.update();
             renderableWorld.generateVBOs();
             secondsSinceLastFrame = 0;
+
+            updateMaterials(); // TODO: this should happen only if necessary, i.e. upon notification from the Asset system.
         }
 
-        if (stereoRenderState != DefaultRenderingProcess.StereoRenderState.MONO) {
+        if (currentRenderingStage != WorldRenderingStage.MONO) {
             playerCamera.updateFrustum();
         }
 
-        // this line needs to be here as deep down it relies on the camera's frustrum.
+        // this line needs to be here as deep down it relies on the camera's frustrum, updated just above.
         renderableWorld.queueVisibleChunks(isFirstRenderingStageForCurrentFrame);
     }
 
     /**
+     * This allows for materials to be swapped at runtime, i.e. if a module provides new ones.
+     */
+    private void updateMaterials() {
+        // TODO: review - perhaps only the chunk shader would ever need to be swapped at runtime? All the others could be final?
+        chunkShader         = Assets.getMaterial("engine:prog.chunk");
+        lightGeometryShader = Assets.getMaterial("engine:prog.lightGeometryPass");
+        //simpleShader        = Assets.getMaterial("engine:prog.simple");  // in use by the currently commented out light stencil pass
+        shadowMapShader     = Assets.getMaterial("engine:prog.shadowMap");
+    }
```

Comment:
The latter should be fine. Certainly will be in the new asset system where this behavior is strongly defined.